sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/README.md

```diff
@@ -1,13 +1,227 @@
+
+# Callback function to add os_type: linux to span properties
+def callback_function(envelope):
+    envelope.data.base_data.properties["os_type"] = "linux"
+    return True
+
+exporter = AzureMonitorSpanExporter(
+    connection_string = os.environ["AZURE_MONITOR_CONNECTION_STRING"]
+)
+exporter.add_telemetry_processor(callback_function)
+
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+span_processor = BatchExportSpanProcessor(exporter)
+trace.get_tracer_provider().add_span_processor(span_processor)
+
+with tracer.start_as_current_span("hello"):
+    print("Hello, World!")
+```
+
+### Modifying Traces
+
+* You can pass a callback function to the exporter to process telemetry before it is exported.
+* Your callback function can return False if you do not want this envelope exported.
+* Your callback function must accept an envelope data type as its parameter.
+* You can see the schema for Azure Monitor data types in the envelopes here.
+* The AzureMonitorSpanExporter handles Data data types.
+
+```Python
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+# Callback function to add os_type: linux to span properties
+def callback_function(envelope):
+    envelope.data.baseData.properties['os_type'] = 'linux'
+    return True
+
+exporter = AzureMonitorSpanExporter(
+    connection_string='InstrumentationKey=<your-ikey-here>'
+)
+# This line will modify telemetry
+exporter.add_telemetry_processor(callback_function)
+
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+span_processor = BatchExportSpanProcessor(exporter)
+trace.get_tracer_provider().add_span_processor(span_processor)
+
+with tracer.start_as_current_span('hello'):
+    print('Hello World!')
+```
+
+### Instrumentation with requests library
+
+OpenTelemetry also supports several instrumentations which allows to instrument with third party libraries.
+
+This example shows how to instrument with the requests_ library.
+
+* Create an Azure Monitor resource and get the instrumentation key, more information can be found here.
+* Install the requests integration package using pip install opentelemetry-ext-http-requests.
+* Specify your connection string in an environment variable `AZURE_MONITOR_CONNECTION_STRING`.
+
+```Python
+import os
+import requests
+from opentelemetry import trace
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+RequestsInstrumentor().instrument()
```

Comment:
I would comment a comment for this line saying something like: `This line causes your calls made with the `requests` library to be tracked`.